azure-kusto-data/azure/kusto/data/_kusto_client.py

```diff
@@ -223,6 +202,135 @@ def _get_table(self, table_id):
     }
 
 
+class _KustoResponseDataSet(six.Iterator):
+    """Represents the parsed data set carried by the response to a Kusto request."""
+
+    def __init__(self):
+        self._tables_names = [t.table_name for t in self.tables]
+        self._index = 0
+        self.tables_count = len(self.tables)
+
+    @property
+    def primary_results(self):
+        if self.tables_count == 1:
+            return self.tables[0]
+        primary = list(
+            filter(lambda x: x.table_kind == WellKnownDataSet.PrimaryResult, self.tables)
+        )
+        if len(primary) == 1:
+            return primary[0]
+        return primary
+
+    @property
+    def errors_count(self):
+        """Checks whether an exception was thrown."""
+        query_status_table = next(
+            (t for t in self.tables if t.table_kind == WellKnownDataSet.QueryCompletionInformation),
+            None,
+        )
+        if not query_status_table:
+            return 0
+        min_level = 4
+        errors = 0
+        for i in range(0, len(query_status_table)):
+            if query_status_table[i][self._error_column] < 4:
+                if query_status_table[i][self._error_column] < min_level:
+                    min_level = query_status_table[i][self._error_column]
+                    errors = 1
+                elif query_status_table[i][self._error_column] == min_level:
+                    errors += 1
+
+        return errors
+
+    def get_exceptions(self):
+        """Gets the excpetions retrieved from Kusto if exists."""
+        query_status_table = next(
+            (t for t in self.tables if t.table_kind == WellKnownDataSet.QueryCompletionInformation),
+            None,
+        )
+        if not query_status_table:
+            return
+        result = []
+        for i in range(0, len(query_status_table)):
+            if query_status_table[i][self._error_column] < 4:
+                result.append(
+                    "Please provide the following data ot Kusto: CRID='{0}' Description:'{1}'".format(
+                        query_status_table[i][self._crid_column],
+                        query_status_table[i][self._status_column],
+                    )
+                )
+        return result
+
+    def __iter__(self):
+        return self
```

Comment:
Looks like you could simply return `iter(self.tables)` here and then this doesn't have to be an iterator?

Reply:
Fixed.